### PR TITLE
[NFC] Rename Heap's T to Element, and iterator to Iterator

### PR DESCRIPTION
**Pull request by airspeedswift**
Similar to #91 but only on internal types.

Motivation:

Simplifies the code by putting the iterator in `Heap`'s namespace,
avoids the two step declare+typealias, and removes freedom of the nested
`Iterator`'s `T` to differ.

Not source-breaking since these are internal types.

Modifications:

Change the placeholder `T` to `Element` and remove now-unnecessary typealias.
Nest `HeapIterator` and rename to `Iterator`, and re-use outer type's `Element`.
Remove a handful of inferrable types.

Result:

Should be a non-functional change.

### CODE REVIEW

**Comment by airspeedswift** [comment]
Drive-by observation while doing this: it looks like [`HeapType`](https://github.com/apple/swift-nio/blob/master/Sources/NIOPriorityQueue/Heap.swift#L21) being public is an accidental leak? It could also be nested, but should probably be marked internal at the same time.

**Comment by Lukasa** [comment]
Yeah, that looks like an accidental leak. Nonetheless, I'm inclined to delay privatising the type.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by airspeedswift** [comment]
If you deprecate the accidentally-public `HeapType`, you could also simplify like this: https://github.com/airspeedswift/swift-nio/commit/3d34d24647f3a4f43d2479f3bc8510ba49b7099c

File: Sources/NIOHTTP1/HTTPTypes.swift [hunk 1]
```diff
@@ -190,7 +161,7 @@ struct HTTPHeadersIterator: IteratorProtocol {
 /// field when needed. It also supports recomposing headers to a maximally joined
 /// or split representation, such that header fields that are able to be repeated
 /// can be represented appropriately.
-public struct HTTPHeaders: Sequence, CustomStringConvertible {
+public struct HTTPHeaders: CustomStringConvertible {
```
Comment (normanmaurer) [inline Sources/NIOHTTP1/HTTPTypes.swift]: @Lukasa I may be wrong but isn't this a breaking change and so we need a next major version ?

Comment (normanmaurer) [reply Sources/NIOHTTP1/HTTPTypes.swift]: ah never mind its just moved.

Comment (airspeedswift) [reply Sources/NIOHTTP1/HTTPTypes.swift]: That's just breaking the conformance out into a separate extension. This is only a style thing but it's generally considered better than conforming in the struct's declaration (this was one of the motivating reasons to make `private` allow access from same-file extensions.

**Comment by normanmaurer** [comment]
@airspeedswift can you rebase ? I will merge after this

**Comment by normanmaurer** [comment]
These were merged into master as (60885f11a8254d31b6c0650892e8e261cde25ee3) and (93f9699802e10a492f0c90a94234d0d7093e8886).

@airspeedswift thanks!

**Pull request closed without merging**